lib/SingleSelleckt.js

```diff
@@ -0,0 +1,776 @@
+'use strict';
+
+var KEY_CODES = require('./KEY_CODES');
+var TEMPLATES = require('./TEMPLATES');
+var templateUtils = require('./templateUtils');
+
+var $ = require('jquery');
+var _ = require('underscore');
+var Mustache = require('Mustache');
+var MicroEvent = require('./MicroEvent');
+
+function SingleSelleckt(options){
+    var settings = _.defaults(options, {
+        mainTemplate: TEMPLATES.SINGLE,
+        itemTemplate: TEMPLATES.SINGLE_ITEM,
+        mainTemplateData: {},
+        selectedClass : 'selected',
+        selectedTextClass : 'selectedText',
+        itemsClass : 'items',
+        itemslistClass : 'itemslist',
+        itemClass : 'item',
+        className : 'dropdown selleckt',
+        highlightClass : 'highlighted',
+        itemTextClass: 'itemText',
+        placeholderText : 'Please select...',
+        enableSearch : false,
+        searchInputClass : 'search',
+        searchThreshold : 0
+    });
+
+    this.$originalSelectEl = options.$selectEl;
+
+    this.mainTemplate = this.mainTemplate || templateUtils.parseTemplate(settings.mainTemplate);
```

Comment:
Aren't `this.mainTemplate` and `this.mainTemplate` below always undefined in the first half of the `||` ? Also `parseTemplate` is called only for its side effects (changing the state of mustache by caching the template).. it feels like it might read better if it's called `cacheTemplate` and returns nothing.
```js
this.mainTemplate = settings.mainTemplate;
this.itemTemplate = settings.itemTemplate;
...

cacheTemplate(this.mainTemplate);
cacheTemplate(this.itemTemplate);
```